amundsen_application/static/js/components/Badges/styles.scss

```diff
@@ -0,0 +1,15 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+@import 'variables';
+
+.clickable-badge-primary {
```

Comment:
For adding the hover styles, those are new specifications within our designs which our components don't have proper support for. It was an unexpected aspect of this ticket that requires some more dedication and what we might refer to as "scope creep". So what I want to propose is that we split your task into two PRs. This one will provide the functionality and the second will be purely UI and CSS focused. What do you think? 

My concern here is just that I didn't intend for your initial tickets to introduce too many concepts at once. I saw this ticket as an opportunity to build a new component, basic css concepts,  and unit tests -- but implementing the overlay as desired may involve specific CSS concepts + additional component restructuring.

Reply:
I will create a new PR to introduce the style changes. I was playing around with this and got something close to what we want so I think it should't be too bad.